### PR TITLE
MINOR: add upgrade guide for Kafka Streams API

### PR DESCRIPTION
**Pull request by mjsax**

### CODE REVIEW

**Comment by mjsax** [comment]
@miguno @guozhangwang @dguy @enothereska

File: docs/upgrade.html [hunk 1]
```diff
@@ -53,6 +53,38 @@
...
 
+<h5><a id="upgrade_1010_streams" href="#upgrade_1010_streams">Streams API changes in 0.10.1.0</a></h5>
+<ul>
+    <li> Stream grouping and aggregation split into two methods: </li>
+    <ul>
+        <li> old: KStream #aggregateByKey(), #reduceByKey(), and #countByKey() </li>
+        <li> new: KStream#groupByKey() plus KGroupedStream #aggregate(), #reduce(), and #count() </li>
+        <li> Example: stream.countByKey() changes to stream.groupByKey().count() </li>
+    </ul>
+    <li> Auto Repartitionig: </li>
```
Comment (miguno) [inline docs/upgrade.html]: Typo: Repartitioning

File: docs/upgrade.html [hunk 2]
```diff
@@ -53,6 +53,38 @@
...
+    <ul>
+        <li> a call to through() after a key-changing operator and before a aggregation/join is not longer required </li>
```
Comment (miguno) [inline docs/upgrade.html]: before [a]n aggregation/join is no[] longer required

File: docs/upgrade.html [hunk 3]
```diff
@@ -53,6 +53,38 @@
...
+        <li> Example: stream.selectKey(...).through(...).countByKey() changes to stream.selectKey().groupByKey().count() </li>
+    </ul>
+    <li> TopologyBuilder: </li>
+    <ul>
+        <li> methods #sourceTopics(String applicationId) and #topicGroups(String applicationId) got simplified to #sourceTopics() and #topicGroups() </li>
+    </ul>
+    <li> DSL: new parameter to specify state store names: </li>
```
Comment (miguno) [inline docs/upgrade.html]: Did you check whether this is valid HTML markup for nested lists?

``` html
<li>...</li>
<ul>
    <li>...</li>
</ul>
```

Shouldn't the `<ul>` be inside the first `<li>...</li>`?

File: docs/upgrade.html [hunk 4]
```diff
@@ -53,6 +53,38 @@
...
+    <ul>
+        <li> New Interactive Queries feature required to specify a store name for all source KTables and window aggregation result KTables (previous parameter "operator/window name" is now the storeName) </li>
```
Comment (miguno) [inline docs/upgrade.html]: "[The] new Interactive Queries feature require[s] specifying a store name for ..."

**Review by miguno — CHANGES_REQUESTED** [review]
Some comments, otherwise LGTM.  Though I think this part of the docs needs more TLC in general.

**Comment by guozhangwang** [comment]
LGTM modulo @miguno 's comments.

File: docs/upgrade.html (hunk 3 above)
Comment (mjsax) [reply docs/upgrade.html]: Good point. You are right.

**Comment by mjsax** [comment]
@miguno @guozhangwang Updated.

**Comment by enothereska** [comment]
LGTM, thanks

**Comment by guozhangwang** [comment]
Merged to trunk and 0.10.1.

**Pull request closed without merging**